### ISSUE TITLE
only load project that is requested

### ISSUE THREAD

**Issue by grosser** [grosser/kennel#172]
atm we always load all projects which takes a few seconds, would be nice to avoid that
autoload parts ... load only project when project.rb project/project.rb project_foo -> project/foo.rb exists

expected savings: ~1s (tried by hardcoding the correct require) (1.6s -> 0.5s to run generate)

we'd need to store a project -> file(s) map to make this work, maybe store after running generate
alternatively it could just support projects with the exact right path and fallback to all for the rest, but given our current messy project layout that would be almost nobody

**Comment by zdrve** [comment]
With the current filtering mechanisms, that is going to rely on introducing a new restriction: that the kennel_id of a project *must* match (1:1 reversible) the class / filename.

**Comment by grosser** [comment]
yeah, that will be hard
we could say "it needs to be either in x.rb or x/x.rb" etc to be flexible but it's not going to be super reliable

**Comment by zdrve** [comment]
Would be perhaps pretty straightforward with the generated_from concept: https://github.com/zdrve/kennel/pull/1

**Comment by grosser** [comment]
yeah that could work as long as the project you are working on was
generated once :)
